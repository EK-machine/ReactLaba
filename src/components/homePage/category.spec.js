/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from "react";
import Category from "./category";

const props = {
  title: "hello",
  icon: "hi",
  path: "/products/pc",
};

describe("Category component tests:", () => {
  it("- Snapshot test", () => {
    const component = shallow(<Category {...props} />);
    expect(component).toMatchSnapshot();
  });

  describe("- Has prop title", () => {
    it("- Should render title", () => {
      const component = shallow(<Category {...props} />);
      const title = component.find(".category__title");
      expect(title.length).toBe(1);
    });

    it("- Expected output of title shall be 'hello'", () => {
      const component = shallow(<Category {...props} />);
      const title = component.find(".category__title");
      expect(title.text()).toBe("hello");
    });
  });
});
