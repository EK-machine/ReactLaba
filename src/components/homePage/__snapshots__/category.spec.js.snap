// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component tests: - Snapshot test 1`] = `
<Link
  className="category__container"
  key="hello"
  to="/products/pc"
>
  <div
    className="category__frame"
  >
    <div
      className="category__icon-container"
    >
      <FontAwesomeIcon
        border={false}
        className="category__icon"
        fixedWidth={false}
        flip={null}
        icon="hi"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
    <div
      className="category__title-container"
    >
      <p
        className="category__title"
      >
        hello
      </p>
    </div>
  </div>
</Link>
`;
